Gems/ScriptCanvas/Code/Editor/Assets/ScriptCanvasFileHandling.cpp

```diff
@@ -77,6 +77,22 @@ namespace ScriptCanvasFileHandlingCpp
     private:
         AZStd::unordered_map<AZ::EntityId, AZ::EntityId> m_newIdsByOld;
     };
+
+    // Create new EntityIds for all EntityIds found in the SC Entity/Component objects
+    // and map all old Ids to the new ones. This way, no Entity activation/deactivation, or
+    // bus communication via EntityId will be handled by multiple or incorrect objects on
+    // possible multiple instantiations of graphs.
+    //
+    // EntityIds contained in variable (those set to self or the graph unique id, will be ignored)
+    void MakeGraphComponentEntityIdsUnique(AZ::Entity* entity, AZ::SerializeContext* serializeContext)
+    {
+        AZStd::unordered_map<AZ::EntityId, AZ::EntityId> remappedIds;
+        remappedIds[AZ::EntityId()] = AZ::EntityId();
+        remappedIds[ScriptCanvas::GraphOwnerId] = ScriptCanvas::GraphOwnerId;
+        remappedIds[ScriptCanvas::UniqueId] = ScriptCanvas::UniqueId;
+
+        AZ::IdUtils::Remapper<AZ::EntityId>::GenerateNewIdsAndFixRefs(entity, remappedIds, serializeContext);
```

Comment:
can you explain what is difference between enumerating objects from serialize context and this method?

Reply:
Enumerating from the serialize context is very slow